docs/keys_and_certificates.md

```diff
@@ -0,0 +1,263 @@
+    - stored in flash and used during attestation
+
+#### Produces
+
+- Fobnail Identity key (signing and encryption)
+    - Fobnail provisioning
+    - used for TLS
+    - key and certificate signed by Platform Owner is stored for later use
+- Fobnail Encryption key (encryption)
+    - Fobnail provisioning
+    - used for storage encryption
+
+### Platform
+
+#### Knows in advance
+
+- root CA for Platform Owner certificate chain
+    - remote platform provisioning
+    - local platform provisioning and attestation - Fobnail Identity key is signed
+    by Platform Owner, so Platform needs a way of verifying trust chain
+- Endorsement Key (encryption)
+    - remote and local platform provisioning
+    - for this project may be considered immutable persistent key unique to TPM
+- EK certificate
+    - EK is unique so is its certificate
+    - _usually_ saved in TPM NVRAM
+    - signed by TPM manufacturer CA
+
+#### Receives
+
+- Platform Owner certificate chain
+    - remote platform provisioning - directly from Platform Owner
+    - local platform provisioning and attestation - from Fobnail Token
+    - used for identification, TLS and verifying Fobnail Identity certificate
+- Fobnail Identity certificate
+    - local platform provisioning and attestation
+    - used for identification and TLS
+
+#### Produces
+
+- Attester Identity Key (signing)
+    - local and remote platform provisioning
+    - created by TPM
+    - trusted based on Make/ActivateCredential and pairing with EK instead of
+    certificate
+
+## Platform Owner certificate chain
+
+Platform owner certificate is used during Fobnail provisioning. Its root CA
+certificate must be preinstalled on Fobnail Token. Due to limited hardware
+capabilities of Fobnail Token, some restrictions apply.
+
+### Maximum certificate chain length
+
+Maximal supported certificate chain consists of 3 certificates: root CA (flashed
+to Fobnail's memory), intermediate CA and final CA that is used to issue
+certificates for keys generated by Fobnail. Intermediate CA is optional, root
+and issuing CA aren't.
+
+### File format
+
+Certificates are concatenated to a single PEM file. Order of concatenation
+matters, certificates must be written starting with leaf certificate and moving
+through intermediate towards root CA certificate. All of them must be present,
+despite the fact that root CA is already known by Fobnail.
+
+### Certificate extensions
+
+Trust chain will be traversed based on `X509v3 Authority Key Identifier`
+extension which is compared against `X509v3 Subject Key Identifier`, so both of
+them must be included in every certificate (former may be skipped for root CA).
+`X509v3 Basic Constraints` is required for CA certificates. `X509v3 Key Usage`
+must specify that all certificates in Platform Owner chain are able to sign
+certificates.
+
+### Validity checks
+
+Fobnail has neither timekeeping nor trusted networking capabilities. Because of
```

Comment:
@krystian-hebel @dpsmith I wonder if this is valid assumption. Can we get trusted time source on Fobnail Token? This seems to be hardware capability - maybe we are limited right now, but we can solve that in the future.

[USB Armory](https://github.com/usbarmory/usbarmory/wiki/Frequently-Asked-Questions-%28FAQ%29#how-do-i-set-the-time) answer to the same problem. Doesn't exist crypto tokens have the same issue? How those USB keys solve that problem?

Reply:
Ok, so this is hardware limitation of nRF52840. IMO above sentence is wrong. Fobnail is not about hardware only. We have to address this problem somewhere in longer run and I see following solutions:
* have more than one reference Fobnail Token for proving Fobnail Project features
* figure out how to obtain trusted time, what should be possible with [NTPsec](https://docs.ntpsec.org/latest/) or something similar

I'm ok with resolving this discussion if we will clearly state that this is current reference Fobnail Token issue. 

@krystian-hebel Also if you see other obstacles in certificates validity checks please let us know, so we can add link that section to release notes.

@dpsmith comments?

Reply:
Maybe give "Validity checks" two subsections, "Temporal Checks" and "Information Checks". Then under "Temporal Checks" that certificate attributes that are of a temporal nature will be limited by a platform's ability to track time. Then under Information Checks, it can be discussed regarding Trust Chain and any other non-temporal certificate attributes.

Thoughts?

Reply:
I've done the split and mentioned that some of restrictions are specific to nRF52840.

Reply:
IMHO it can be resolved.